power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -69,8 +78,10 @@ inline UpdateChange update_component(MainModelState<ComponentContainer>& state,
     detail::iterate_component_sequence<Component>(
         [&state_changed, &changed_it, &state](UpdateType const& update_data, Idx2D const& sequence_single) {
             auto& comp = get_component<Component>(state, sequence_single);
+#ifdef _DEBUG
```

Comment:
you don't need this one. `assert` is only enabled in debug build. You don't need a separate macro.

Reply:
linux and macos complained about `_DEBUG`

Reply:
They still do ;(